libraries/chain/webassembly/runtimes/eos-vm-oc/LLVMJIT.cpp

```diff
@@ -305,14 +305,19 @@ namespace LLVMJIT
 
 		unsigned num_functions_stack_size_found = 0;
 		for(const auto& stacksizes : jitModule->unitmemorymanager->stack_sizes) {
-			llvm::DataExtractor ds(llvm::ArrayRef(stacksizes.data(), stacksizes.size()), true, 8);
-			llvm::DataExtractor::Cursor c(0);
-
-			while(!ds.eof(c)) {
-				ds.getAddress(c);
-				WAVM_ASSERT_THROW(!!c);
-				const uint64_t stack_size = ds.getULEB128(c);
-				WAVM_ASSERT_THROW(!!c);
+#if LLVM_VERSION_MAJOR < 10
+			using de_offset_t = uint32_t;
+#else
+			using de_offset_t = uint64_t;
+#endif
+			llvm::DataExtractor ds(llvm::StringRef(reinterpret_cast<const char*>(stacksizes.data()), stacksizes.size()), true, 8);
```

Comment:
okay okay, I know this looks all sorts of ick, but take a look at,
https://github.com/llvm/llvm-project/blob/release/11.x/llvm/include/llvm/Support/DataExtractor.h#L85
The ctor that takes an `ArrayRef` does the exact same cast + `StringRef` as I'm doing here. So this is not really even a change for existing working versions.